### PR TITLE
Drip when collecting

### PR DESCRIPTION
**Pull request by CodeSandwich**
The second step toward adding drips. Depends on https://github.com/radicle-dev/radicle-streaming/pull/31

### CODE REVIEW

File: src/Pool.sol [hunk 1]
```diff
@@ -213,51 +216,93 @@ abstract contract Pool {
...
+            collected -= dripped;
+            uint128 drippedPerWeight = dripped / sender.weightSum;
+            // Iterating over receivers, see `ReceiverWeights` for details
+            address dripsAddr = ReceiverWeightsImpl.ADDR_ROOT;
+            address hint = ReceiverWeightsImpl.ADDR_ROOT;
+            while (true) {
+                uint32 weight;
+                (dripsAddr, hint, weight) = sender.receiverWeights.nextWeight(dripsAddr, hint);
+                if (dripsAddr == ReceiverWeightsImpl.ADDR_ROOT) break;
+                receivers[dripsAddr].collectable += drippedPerWeight * weight;
```
Comment (manuelpolzhofer) [inline src/Pool.sol:303]: Should  we add some sanity checks? I already checked `weightSum` seems correct, etc.

We could track the `totalDripped` here in the loop and afterwards 
```
require(dripped == totalDdripped)
```

At least for stress testing it, would be maybe nice to have it.

File: src/Pool.sol [hunk 2]
```diff
@@ -213,51 +216,93 @@ abstract contract Pool {
     /// @notice Returns amount of received funds available for collection
     /// @param receiverAddr The address of the receiver
     /// @return collected The available amount
-    function collectable(address receiverAddr) public view returns (uint128) {
+    function collectable(address receiverAddr)
+        public
+        view
+        returns (uint128 collected, uint128 dripped)
```
[the remark this answers is not in the document]
Comment (CodeSandwich) [reply src/Pool.sol:222]: Isn't `totalCollected == kept + dripped`? It's not very useful and it adds even more elements to the API with additional rules.

File: src/Pool.sol (hunk 1 above)
Comment (CodeSandwich) [reply src/Pool.sol:303]: I'm a bit skeptical about usefulness of adding `require(dripped == totalDripped)` to production code for sake of stress testing. We must test if everybody actually gets what they're supposed to get and if `dripped != totalDripped`, these checks will catch that and many more issues.

Comment (CodeSandwich) [reply src/Pool.sol]: Added the sanity check

Comment (manuelpolzhofer) [reply src/Pool.sol]: Okay nice. However, if we run into a problem and it reverts the fund would be locked.

File: src/Pool.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (manuelpolzhofer) [reply src/Pool.sol:229]: Yes, no strong opinion of adding it.

**Review by manuelpolzhofer — APPROVED** [review]

**Pull request merged**